### PR TITLE
Added windows support + changed file structure

### LINKED ISSUE #2: Windows support
Add the pre-built files (no headers - they should be shared between all the operating systems) for windows.
Update the build.cs to support this platform

### PR DESCRIPTION
**Pull request by EliLentz**
Added in this PR:
--

* Added windows libraries (.lib + .dll)
* Addition of hiredis and redis++ header files specifically tailored for Windows.
* Replaced Linux header files
* Added Windows support to .build file
* Changed .gitignore (removed *.lib)

### CODE REVIEW

File: Source/URedis/URedis.Build.cs
Comment (EliLentz) [inline Source/URedis/URedis.Build.cs:1]: This file was auto corrected by Visual Studio

Comment (galeone) [reply Source/URedis/URedis.Build.cs:1]: I have created the .clang-format and the pre-commit for a reason. configure visual studio to apply our coding convention and also install and execute pre-commit. In this way the rules will be automatically applied on every commit

Comment (galeone) [reply Source/URedis/URedis.Build.cs:1]: The binary files should be tracked with git lfs. I try to fix this myself switching in the branch you created

**Comment by galeone** [comment]
The binary files should be tracked with git lfs. I try to fix this myself switching in the branch you created.

Also, there are no .lib committed

**Comment by galeone** [comment]
I fixed the tracking of the .dll committed - but there are not .lib committed. Did you forget them?

I executed the pre-commit `pre-commit run --all` to fix the wrongly formatted build.cs files.

**Comment by galeone** [comment]
Anyway, I noticed that you uploaded different libraries (and thus header) for windows. This is a problem for the future maintenance. If we want to offer the same functionalities in all the platform, we should use the same libraries at the same version in all of them.

e.g.
```diff
diff -r --color linux/hiredis/hiredis.h windows/hiredis/hiredis.h
42c42
< typedef long long ssize_t;
---
> typedef intptr_t ssize_t;
49,51c49,51
< #define HIREDIS_MINOR 1
< #define HIREDIS_PATCH 1
< #define HIREDIS_SONAME 1.1.1-dev
---
> #define HIREDIS_MINOR 2
> #define HIREDIS_PATCH 0
> #define HIREDIS_SONAME 1.1.0
diff -r --color linux/hiredis/sds.h windows/hiredis/sds.h
```

Since you uploaded a never version of the libraries, I'll update the linux lib as well (I'll do it in this branch whenever I have a bit of time)

File: Source/URedis/URedis.Build.cs
Comment (EliLentz) [reply Source/URedis/URedis.Build.cs:1]: Ok. Thank you for the devtool addition (I saw them)
But, I have never worked with Clang-Format (Im working with C#, C++ is just my hobby😊) and Github as version controller (I used to use Azure).
So, I'll read more about Clang-Format

**Comment by galeone** [comment]
I noticed another thing that needs to be fixed. You compiled redis-plus-plus somehow setting the standard to something lower than C++17.

Please recompile it forcing the standard to be C++17 (you should configure this through cmake). I noticed because the windows headers are too different from the linux headers I uploaded  - in fact, it's re-implementing some basic C++17 features like string views and optionals

**Comment by EliLentz** [comment]
> I noticed another thing that needs to be fixed. You compiled redis-plus-plus somehow setting the standard to something lower than C++17.
> 
> Please recompile it forcing the standard to be C++17 (you should configure this through cmake). I noticed because the windows headers are too different from the linux headers I uploaded - in fact, it's re-implementing some basic C++17 features like string views and optionals

I've added C++ 17 standart for compiling to .build file and it broke the pre-commit check :(

**Comment by galeone** [comment]
That's wrong. I was talking about the standard used to compile the Windows library (outside this unreal project) - not the standard defined in the build.cs. I revert this change

Btw, read this to understand the pre-commit part: https://pre-commit.com/

**Comment by galeone** [comment]
@EliLentz I downgraded the Linux libraries and added a readme where it's clear the version we use in Linux/Windows and any other OS. I compiled the libraries setting C++11 also on Linux, this way there's nothing you have to do.

Please just try this branch locally and check if the library works on Windows. If it does, this can be merged (verified and this branch works on Linux of course)

**Pull request merged**